server/src/main/java/io/unitycatalog/server/persist/dao/CommitDAO.java

```diff
@@ -0,0 +1,79 @@
+package io.unitycatalog.server.persist.dao;
+
+import io.unitycatalog.server.model.Commit;
+import io.unitycatalog.server.model.CommitInfo;
+import jakarta.persistence.*;
+import java.util.Date;
+import java.util.UUID;
+import lombok.*;
+import org.hibernate.annotations.UuidGenerator;
+
+@Entity
+@Table(
+    name = "uc_commits",
+    uniqueConstraints = {
+      @UniqueConstraint(columnNames = {"metastore_id", "table_id", "commit_version"})
+    })
+// Lombok
+@Getter
+@Setter
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class CommitDAO {
+  @Id
+  @UuidGenerator
+  @Column(name = "id", columnDefinition = "BINARY(16)")
```

Comment:
Why do we need id here?

Reply:
This is primarily to simplify the primary key, it could be replaced with @IdClass or @EmbeddedId to have (table_id, version) as the primary key but it would add complexity.